### PR TITLE
JWT cleanup

### PR DESCRIPTION
**Pull request by jayteemo**
Initialize JWT buffer to NULL so that the library allocates a properly sized buffer for the JWT.
Ensure JWT has base64url format.
CIA-424 and CIA-425

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

File: lib/multicell_location/services/nrf_cloud_integration.c [hunk 1]
```diff
@@ -101,6 +104,7 @@ int location_service_get_cell_location(const struct lte_lc_cells_info *cell_data
 	err = nrf_cloud_rest_cell_pos_get(&rest_ctx, &loc_req, &result);
 
 	modem_jwt_free(jwt.jwt_buf);
+	jwt.jwt_buf = NULL;
```
Comment (plskeggs) [inline lib/multicell_location/services/nrf_cloud_integration.c:107]: I didn't have this change.  Is this just defensive?

Unfortunately the other changes in this PR are insufficient by themselves to fix the problem.

Comment (jayteemo) [reply lib/multicell_location/services/nrf_cloud_integration.c:107]: > Is this just defensive?
Yep.

**Review by peknis — APPROVED** [review]

File: lib/multicell_location/services/nrf_cloud_integration.c [hunk 2]
```diff
@@ -75,7 +75,10 @@ int location_service_get_cell_location(const struct lte_lc_cells_info *cell_data
 		.exp_delta_s = (5 * 60),
 		.sec_tag = CONFIG_MULTICELL_LOCATION_NRF_CLOUD_JWT_SEC_TAG,
 		.key = JWT_KEY_TYPE_CLIENT_PRIV,
-		.alg = JWT_ALG_TYPE_ES256
+		.alg = JWT_ALG_TYPE_ES256,
+		/* Set to NULL so a properly sized buffer is allocated */
+		.jwt_buf = NULL,
+		.jwt_sz = 0
```
Comment (SeppoTakalo) [inline lib/multicell_location/services/nrf_cloud_integration.c:81]: I believe unlisted members are initialized to zero, if a shorter initializer list is given.

but it does not harm to be verbal about it.

**Review by SeppoTakalo — APPROVED** [review]

File: lib/modem_jwt/modem_jwt.c [hunk 3]
```diff
@@ -60,6 +60,31 @@ static int parse_jwt_at_cmd_resp(char *const jwt_resp)
...
+		*found = '_';
+	}
+
+	/* remove padding '=' */
+	found = strchr(base64_string, '=');
+	if (found) {
+		*found = '\0';
+	}
+}
+
```
Comment (SebastianBoe) [inline lib/modem_jwt/modem_jwt.c:87]: Use a while-loop next time.

**Review by SebastianBoe — APPROVED** [review]

**Pull request merged**